.github/workflows/haskell.yml

```diff
@@ -36,7 +36,7 @@ jobs:
   build:
     strategy:
       matrix:
-        ghc: ["8.10.7", "9.2.8", "9.6.6", "9.8.4", "9.10.1"]
+        ghc: ["8.10.7", "9.2.8", "9.6.7", "9.8.4", "9.10.1", "9.12.1"]
```

Comment:
You need to make the same change in the test matrix further down. This is why CI is failing for all the 9.6.6 jobs.

Reply:
Done!

Reply:
@neilmayhew It still fails in the same way. The message is:

> Run actions/download-artifact@v4
Downloading single artifact
Error: Unable to download artifact(s): Artifact not found for name: state-9.6.6-ubuntu-latest
        Please ensure that your artifact is not expired and the artifact was uploaded using a compatible version of toolkit/upload-artifact.
        For more information, visit the GitHub Artifacts FAQ: https://github.com/actions/toolkit/blob/main/packages/artifact/docs/faq.md

Reply:
That is the change that you are missing: https://github.com/IntersectMBO/cardano-ledger/pull/4979/files#r2038939056